### PR TITLE
Add Implicit ARIA roles

### LINKED ISSUE #22699: Show *Implicit ARIA Role* on each `<input type="*">` page
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input

### What specific section or headline is this issue about?

/Web/HTML/Element/input/*

### What information was incorrect, unhelpful, or incomplete?

The `<input type="*">` pages are missing information about their implied ARIA role. 

The implicit ARIA roles are described on the `<input>` page, which is good. Unfortunately, if a reader doesn't think to go up a level from the type pages, they won't have access to this information.

### What did you expect to see?

There's a table on each input type page. I expect implied ARIA role to be one of those rows.

<table class="properties">
  <tbody>
    <tr>
      <td><strong><a href="#value">Value</a></strong></td>
      <td>
        A string representing the value contained in
        the search field.
      </td>
    </tr>
    <tr>
      <td><strong>Events</strong></td>
      <td>
        {{domxref("HTMLElement/change_event", "change")}} and
        {{domxref("HTMLElement/input_event", "input")}}
      </td>
    </tr>
    <tr>
      <td><strong>Supported Common Attributes</strong></td>
      <td>
        {{htmlattrxref("autocomplete", "input")}},
        {{htmlattrxref("list", "input")}},
        {{htmlattrxref("maxlength", "input")}},
        {{htmlattrxref("minlength", "input")}},
        {{htmlattrxref("pattern", "input")}},
        {{htmlattrxref("placeholder", "input")}},
        {{htmlattrxref("required", "input")}},
        {{htmlattrxref("size", "input")}}.
      </td>
    </tr>
    <tr>
      <td><strong>IDL attributes</strong></td>
      <td><code>value</code></td>
    </tr>    
    <tr>
      <td><strong>Implicit ARIA roles</strong></td>
      <td>
            <ul>
              <li>
                with no <code>list</code> attribute: {{ARIARole("searchbox")}}
              </li>
              <li>
                with <code>list</code> attribute:{{ARIARole("combobox")}}
              </li>
            </ul>
      </td>
    </tr>
    <tr>
      <td><strong>DOM interface</strong></td>
      <td><p>{{domxref("HTMLInputElement")}}</p></td>
    </tr>
    <tr>
      <td><strong>Methods</strong></td>
      <td>
        {{domxref("HTMLInputElement.select", "select()")}},
        {{domxref("HTMLInputElement.setRangeText", "setRangeText()")}},
        {{domxref("HTMLInputElement.setSelectionRange", "setSelectionRange()")}}.
      </td>
    </tr>
  </tbody>
</table>

### Do you have any supporting links, references, or citations?

[Input page](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input)
[Example Input Type page](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search) (one of many)

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/html/element/input`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/html/element/input/index.md
* Last commit: https://github.com/mdn/content/commit/8e16e19d3ee35157c81b299c0ccb9d5effd2ac22
* Document last modified: 2022-11-26T11:30:26.000Z

</details>

### PR DESCRIPTION
**Pull request by dawei-wang**
Fix mdn/content#22699

### CODE REVIEW

File: files/en-us/web/html/element/input/image/index.md [hunk 1]
```diff
@@ -332,6 +332,10 @@ Here, `object-position` is configured to draw the image in the top-right corner
       <td><strong>Methods</strong></td>
       <td>None.</td>
     </tr>
+     <tr>
+      <td><strong>Implicit ARIA Role</strong></td>
+      <td><a href="/en-US/docs/Web/Accessibility/ARIA/Roles/button_role"><code>button</code></a></td>
```
Comment (estelle) [inline files/en-us/web/html/element/input/image/index.md:337]: {{ARIARole("button")}}?

File: files/en-us/web/html/element/input/email/index.md [hunk 2]
```diff
@@ -378,6 +378,12 @@ As an added touch, the {{HTMLElement("label")}} element is used to establish a l
         {{domxref("HTMLInputElement.select", "select()")}}
       </td>
     </tr>
+    <tr>
+      <td><strong>Implicit ARIA Role</strong></td>
+      <td>with no <code>list</code> attribute:
+                <code><a href="/en-US/docs/Web/Accessibility/ARIA/Roles/textbox_role">textbox</a></code></td>
```
Comment (estelle) [inline files/en-us/web/html/element/input/email/index.md:384]: {{ARIARole("textbox")}}?

**Review by estelle — COMMENTED** [review]
should we be using `{{ARIARole("ROLE")}}` or `<a link>role</a>`

**Comment by dawei-wang** [comment]
> should we be using `{{ARIARole("ROLE")}}` or `<a link>role</a>`

@estelle Good question. I was wondering the same thing.  The [Input subpage technical summary](/en-US/docs/Web/HTML/Element/input#technical_summary) uses both.

File: files/en-us/web/html/element/input/email/index.md [hunk 3]
```diff
@@ -378,6 +378,12 @@ As an added touch, the {{HTMLElement("label")}} element is used to establish a l
...
+      <td>with <code>list</code> attribute: {{ARIARole("combobox")}}</td>
```
Comment (dawei-wang) [inline files/en-us/web/html/element/input/email/index.md:385]:
```suggestion
      <td>with <code>list</code> attribute: <code><a href="/en-US/docs/Web/Accessibility/ARIA/Roles/combobox_role">combobox</a></code></td>
```

File: files/en-us/web/html/element/input/number/index.md [hunk 4]
```diff
@@ -445,6 +445,12 @@ Also consider using the [`autocomplete`](/en-US/docs/Web/HTML/Attributes/autocom
         {{domxref("HTMLInputElement.stepDown", "stepDown()")}}
       </td>
     </tr>
+    <tr>
+      <td><strong>Implicit ARIA Role</strong></td>
+      <td>
+        {{ARIARole("spinbutton")}}
```
Comment (dawei-wang) [inline files/en-us/web/html/element/input/number/index.md:451]:
```suggestion
        <code><a href="/en-US/docs/Web/Accessibility/ARIA/Roles/spinbutton_role">spinbutton</a></code>
```

File: files/en-us/web/html/element/input/radio/index.md [hunk 5]
```diff
@@ -336,6 +336,12 @@ Notice that when clicking on a radio button, there's a nice, smooth fade out/in
         {{domxref("HTMLInputElement.select", "select()")}}
       </td>
     </tr>
+     <tr>
+      <td><strong>Implicit ARIA Role</strong></td>
+      <td>
+        {{ARIARole("radio")}}
```
Comment (dawei-wang) [inline files/en-us/web/html/element/input/radio/index.md:342]:
```suggestion
        <code><a href="/en-US/docs/Web/Accessibility/ARIA/Roles/radio_role">radio</a></code>
```

File: files/en-us/web/html/element/input/range/index.md [hunk 6]
```diff
@@ -391,6 +391,12 @@ input[type="range"][orient="vertical"] {
         and {{domxref("HTMLInputElement.stepUp", "stepUp()")}}
       </td>
     </tr>
+    <tr>
+      <td><strong>Implicit ARIA Role</strong></td>
+      <td>
+        {{ARIARole("slider")}}
```
Comment (dawei-wang) [inline files/en-us/web/html/element/input/range/index.md:397]:
```suggestion
        <code><a href="/en-US/docs/Web/Accessibility/ARIA/Roles/slider_role">slider</a></code>
```

File: files/en-us/web/html/element/input/search/index.md [hunk 7]
```diff
@@ -454,6 +454,11 @@ You can see a good example of a search form used in context at our [website-aria
         {{domxref("HTMLInputElement.setSelectionRange", "setSelectionRange()")}}.
       </td>
     </tr>
+     <tr>
+      <td><strong>Implicit ARIA Role</strong></td>
+      <td>with no <code>list</code> attribute: {{ARIARole("searchbox")}}</td>
+      <td>with <code>list</code> attribute: {{ARIARole("combobox")}}</td>
```
Comment (dawei-wang) [inline files/en-us/web/html/element/input/search/index.md:460]:
```suggestion
      <td>with no <code>list</code> attribute: <code><a href="/en-US/docs/Web/Accessibility/ARIA/Roles/searchbox_role">searchbox</a></code></td>
      <td>with <code>list</code> attribute: <code><a href="/en-US/docs/Web/Accessibility/ARIA/Roles/combobox_role">combobox</a></code></td>
```

File: files/en-us/web/html/element/input/tel/index.md [hunk 8]
```diff
@@ -522,6 +470,66 @@ input:valid + span::after {
...
+        {{domxref("HTMLInputElement.select", "select()")}},
+        {{domxref("HTMLInputElement.setRangeText", "setRangeText()")}},
+        {{domxref("HTMLInputElement.setSelectionRange", "setSelectionRange()")}}
+      </td>
+    </tr>
+    <tr>
+      <td><strong>Implicit ARIA Role</strong></td>
+      <td>with no <code>list</code> attribute:
+                <code><a href="/en-US/docs/Web/Accessibility/ARIA/Roles/textbox_role">textbox</a></code></td>
+      <td>with <code>list</code> attribute: {{ARIARole("combobox")}}</td>
```
Comment (dawei-wang) [inline files/en-us/web/html/element/input/tel/index.md:528]:
```suggestion
      <td>with <code>list</code> attribute: <code><a href="/en-US/docs/Web/Accessibility/ARIA/Roles/combobox_role">combobox</a></code></td>
```

File: files/en-us/web/html/element/input/text/index.md [hunk 9]
```diff
@@ -432,6 +432,12 @@ You can see good examples of text inputs used in context in our [Your first HTML
         {{domxref("HTMLInputElement.setSelectionRange", "setSelectionRange()")}}.
       </td>
     </tr>
+    <tr>
+      <td><strong>Implicit ARIA Role</strong></td>
+      <td>with no <code>list</code> attribute:
+                <code><a href="/en-US/docs/Web/Accessibility/ARIA/Roles/textbox_role">textbox</a></code></td>
+      <td>with <code>list</code> attribute: {{ARIARole("combobox")}}</td>
```
Comment (dawei-wang) [inline files/en-us/web/html/element/input/text/index.md:439]:
```suggestion
      <td>with <code>list</code> attribute: <code><a href="/en-US/docs/Web/Accessibility/ARIA/Roles/combobox_role">combobox</a></code></td>
```

File: files/en-us/web/html/element/input/url/index.md [hunk 10]
```diff
@@ -398,6 +398,12 @@ You can also find our [pattern validation example on GitHub](https://github.com/
         {{domxref("HTMLInputElement.setSelectionRange", "setSelectionRange()")}}.
       </td>
     </tr>
+    <tr>
+      <td><strong>Implicit ARIA Role</strong></td>
+      <td>with no <code>list</code> attribute:
+                <code><a href="/en-US/docs/Web/Accessibility/ARIA/Roles/textbox_role">textbox</a></code></td>
+      <td>with <code>list</code> attribute: {{ARIARole("combobox")}}</td>
```
Comment (dawei-wang) [inline files/en-us/web/html/element/input/url/index.md:405]:
```suggestion
      <td>with <code>list</code> attribute: <code><a href="/en-US/docs/Web/Accessibility/ARIA/Roles/combobox_role">combobox</a></code></td>
```

**Review by dawei-wang — COMMENTED** [review]
Used a href  link since the {{ARIARole}} macro is deprecated per https://github.com/mdn/yari/pull/8025

**Review by estelle — APPROVED** [review]
thank you

**Pull request merged**